content/browser/ml/webnn/dml/graph_dml_impl.cc

```diff
@@ -1052,6 +1053,15 @@ void GraphDMLImpl::AddArgMinMax(OperatorType operator_type,
 
   auto node_output = graph_desc_builder_->CreateNodeOutput(
       node, 0, std::move(output_tensor_desc));
+
+  if (!keep_dimensions) {
```

Comment:
Propose the following, just checking `original_output_tensor_desc` and calling CreateNodeOutput once rather than overwriting a previous call.

```
  // Determine output sizes. Ignore output_desc->dimensions for the dimensions,
  // since DirectML expects the output dimensions to have the same rank as the
  // input, and output_desc->dimensions may have removed dimensions if
  // keepDimensions was false.
  std::vector<uint32_t> output_dimensions = input_dimensions;
  DCHECK(axis < output_dimensions.size());
  output_dimensions[axis] = 1u;
  auto output_data_type = GetTensorDataType(output_desc->data_type);
  TensorDesc output_tensor_desc(output_data_type, output_dimensions);
  TensorDesc original_output_tensor_desc(output_data_type,  //<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
                                         output_desc->dimensions);

  // DML accepts multiple axes. So pass the single index along.
  std::array<uint32_t, 1> axes = {axis};

  // Note DML_ARGMIN_OPERATOR_DESC and DML_ARGMAX_OPERATOR_DESC are
  // identical in structure layout. So we can use for either.
  DML_ARGMIN_OPERATOR_DESC operator_desc = {};
  operator_desc.InputTensor = input_tensor_desc.Get();
  operator_desc.OutputTensor = output_tensor_desc.Get();
  operator_desc.AxisCount = 1u;
  operator_desc.Axes = axes.data();
  operator_desc.AxisDirection =
      static_cast<DML_AXIS_DIRECTION>(select_last_index);

  DML_OPERATOR_TYPE dml_operator_type = DML_OPERATOR_INVALID;
  switch (operator_type) {
    case OperatorType::kArgMin:
      dml_operator_type = DML_OPERATOR_ARGMIN;
      break;
    case OperatorType::kArgMax:
      dml_operator_type = DML_OPERATOR_ARGMAX;
      break;
    default:
      NOTREACHED();
  }
  Node node = graph_desc_builder_->CreateOperatorNode(dml_operator_type,
                                                      &operator_desc);

  graph_desc_builder_->Connect({input_node}, {node});

  auto node_output = graph_desc_builder_->CreateNodeOutput(
      node, 0, std::move(original_output_tensor_desc)); //<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
  node_output_map_[output_index] = std::move(node_output);
```

Reply:
Thanks, updated, PTAL! @fdwr